evalml/tests/component_tests/test_lgbm_classifier.py

```diff
@@ -328,3 +329,46 @@ def test_lgbm_preserves_schema_in_rename(mock_predict_proba, mock_predict, mock_
     assert mock_predict.call_args[0][0].ww.schema == original_schema
     lgb.predict_proba(X)
     assert mock_predict_proba.call_args[0][0].ww.schema == original_schema
+
+
+def test_lgbm_handle_nullable_types(
+    nullable_type_test_data,
+    nullable_type_target,
+    split_nullable_logical_types_by_compatibility,
+):
+    X = nullable_type_test_data(has_nans=False)
+    X = X.ww.select(include=["numeric", "Boolean", "BooleanNullable"])
+
+    lgb = LightGBMClassifier()
+
+    (
+        compatible_y_ltypes,
+        incompatible_y_ltypes,
+    ) = split_nullable_logical_types_by_compatibility(
+        "y" in lgb._integer_nullable_incompatibilities,
+        "y" in lgb._boolean_nullable_incompatibilities,
+    )
+
+    for nullable_ltype in incompatible_y_ltypes + compatible_y_ltypes:
+        y = nullable_type_target(ltype=nullable_ltype, has_nans=False)
+
+        if nullable_ltype in incompatible_y_ltypes:
+            with pytest.raises(
+                ValueError,
+                match=re.escape(
+                    "Unknown label type: 'unknown'",
+                ),
+            ):
+                lgb.fit(X, y)
+
+            # Confirm using the handle method lets the transform work
+            X_d, y_d = lgb._handle_nullable_types(X, y)
+            lgb.fit(X_d, y_d)
+            lgb.predict(X_d)
+        else:
+            lgb.fit(X, y)
+            lgb.predict(X)
+            lgb.predict_proba(X)
```

Comment:
Mega nit: can you simplify this a bit by setting `X, y = lgb._handle_nullable_types(X, y)` and calling fit, predict, and predict_proba outside of the conditional block?